doc/specs/#1235 - Azure cloud shell connector/spec.md

```diff
@@ -0,0 +1,78 @@
+---
+author: Pankaj Bhojwani pankaj.d.bhoj@gmail.com
+created on: 2019-06-12
+last updated: 2019-06-12
+issue id: #1235
+---
+
+# Azure cloud shell connector
+
+## Abstract
+
+This spec goes over the details of how a feature enabling Windows Terminal users to connect to the Azure cloud shell should behave. It includes implementation and design considerations. 
+
+## Inspiration
+
+The idea is to give developers access to their Azure services smoothly within the Windows Terminal app, letting them engage with Azure technologies in a convenient manner. By integrating the Azure cloud shell into Windows Terminal, we can do just that. 
+
+## Solution Design
+
+The flowchart below shows the process by which the Azure cloud shell will be integrated into Windows Terminal. 
+
+![Sol Design](images/solDesign.png)
+
+The first three steps - authenticating the user, requesting a cloud shell and requesting a terminal - will be done via http requests. These requests will use the [cpprestsdk](https://github.com/Microsoft/cpprestsdk) library as that library is also owned by Microsoft, making it easy to resolve issues should any arise. 
+
+Authenticating the user will use [device code flow](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Device-Code-Flow) since Windows Terminal does not support browser access (yet). As for the authentication endpoint, Azure AD v1.0 will be used because Azure AD v2.0 (also known as Microsoft Identity Platform) [does not support login to personal accounts with device code flow](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Device-Code-Flow#constraints) at this time. Furthermore, upon successful authentication, the login/token information will be stored so that users will not need to repeatedly go through device code flow for future logins. Since this is sensitive information, the tokens will be stored with [Windows Storage](https://docs.microsoft.com/en-us/uwp/api/windows.storage) and encrypted with [Windows Security Data Protection](https://docs.microsoft.com/en-us/uwp/api/windows.security.cryptography.dataprotection.dataprotectionprovider). 

```

Comment:
Is it too onerous to launch another window that has nothing but a web host in it? That's the flow I'm familiar with from tons of other applications doing an authentication.

Reply:
For text-only applications like this, it is recommended to use device code flow - see [here](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Device-Code-Flow) and [here ](https://auth0.com/docs/flows/concepts/device-auth).